Automattic/MobilePay-Apple
# Networking with URLSession

## Description

Resolves https://github.com/wordpress-mobile/WordPress-iOS/issues/16863

This PR adds the initial networking layer:

- Added `InAppPurchasesService` to call the fetch product skus / create order endpoints
- Added initial `InAppPurchasesServiceTests`
- Added `InAppPurchasesAPIRouter` to organize the endpoints 

Endpoints ref: pdekpA-18-p2

## Notes
- This PR is a combination of two earlier PRs
    - https://github.com/Automattic/MobilePay-Apple/pull/5 (approved)
    - https://github.com/Automattic/MobilePay-Apple/pull/7 

## How to test 

1. Run the iOS app
2. Tap on Rocket Fuel to purchase
3. Tap Confirm
4. Tap OK
5. On Xcode debugger menu, click on the Manage StoreKit Transactions menu item 
6. ✅ There should be a record of Rocket Fuel being purchased (See example below)

<img width="747" alt="Screen Shot 2021-07-19 at 16 32 32" src="https://user-images.githubusercontent.com/6711616/126186582-3f76559e-7fe5-45ae-9068-2754883f6351.png">
